core/bin/qdrant_migrator.rs

```diff
@@ -0,0 +1,434 @@
+                            "[DANGER] Are you sure you want to delete this qdrant \
+                             shadow_write_cluster collection? \
+                             (this is definitive) shadow_write_cluster={} points_count={}",
+                            match qdrant_clients.shadow_write_cluster(&ds.config().qdrant_config) {
+                                Some(cluster) => cluster.to_string(),
+                                None => "none".to_string(),
+                            }
+                            .to_string(),
+                            info.points_count,
+                        ))? {
+                            true => (),
+                            false => Err(anyhow!("Aborted"))?,
+                        }
+                    }
+                    None => Err(anyhow!("Qdrant collection not found"))?,
+                };
+
+                shadow_write_qdrant_client
+                    .delete_collection(ds.qdrant_collection())
+                    .await?;
+
+                utils::done(&format!(
+                    "Deleted qdrant shadow_write_cluster collection: \
+                     collection={} shadow_write_cluster={}",
+                    ds.qdrant_collection(),
+                    match qdrant_clients.shadow_write_cluster(&ds.config().qdrant_config) {
+                        Some(cluster) => cluster.to_string(),
+                        None => "none".to_string(),
+                    }
+                ));
+
+                // Remove shadow_write_cluster from config.
+                let mut config = ds.config().clone();
+
+                config.qdrant_config = match config.qdrant_config {
+                    Some(c) => Some(QdrantDataSourceConfig {
+                        cluster: c.cluster,
+                        shadow_write_cluster: None,
+                    }),
+                    None => Some(QdrantDataSourceConfig {
+                        cluster: QdrantCluster::Main0,
+                        shadow_write_cluster: None,
+                    }),
+                };
+
+                ds.update_config(store, &config).await?;
+
+                utils::done(&format!(
+                    "Updated data source: collection={} cluster={} shadow_write_cluster={}",
+                    ds.qdrant_collection(),
+                    qdrant_clients
+                        .main_cluster(&ds.config().qdrant_config)
+                        .to_string(),
+                    match qdrant_clients.shadow_write_cluster(&ds.config().qdrant_config) {
+                        Some(cluster) => cluster.to_string(),
+                        None => "none".to_string(),
+                    }
+                ));
+
+                Ok::<(), anyhow::Error>(())
+            }
+            Commands::MigrateShadowWrite {
+                project_id,
+                data_source_id,
+            } => {
+                // This is the most dangerous command of all as it is the only one to actually
```

Comment:
I think this comment is wrong

Reply:
It's accurate

Reply:
isn't the `clear` command the most dangerous one ?